src/puppetlabs/puppetdb/query/paging.clj

```diff
@@ -113,8 +115,34 @@
                  "unknown key '" (name (first (keys (dissoc bad-order-by :field :order)))) "'.")))
     order_by))
 
+(defn parse-order-by'
+  "Given a list of order-by clauses that conform to the PuppetDB
+  paging API, validate and convert the order-by to our internal
+  format. Incoming is of the form:
+
+  {:field String (s/optional-key :order) String}
+
+  outgoing is of the form:
+
+  [(s/one s/Keyword \"field-name\") (s/one (s/enum :ascending :descending) \"sort-order\")]
+
+  Not applying schema here until the original parse-order-by fn is
+  replaced by this one. There are tests that require bad input
+  parameters to be handled properly inside this function
+
+  If validation fails, this function will throw an exception with
+  an informative error message as to the cause of the failure."
+  [order-by]
+  {:post [(every? order-by-expr? %)]}
+  (when order-by
+    (->> order-by
+         validate-order-by-data-structure
+         validate-no-invalid-order-by-fields
+         parse-required-order-by-fields)))
+
 (defn parse-order-by
-  "Given a map of paging-options, validates that the order_by field conforms to
+  "DEPRECATED - this function will be replaced by parse-order-by'
```

Comment:
The pattern I've followed here to minimize change is to create a parallel function that works on the query map rather than the request map. When possible I've been making the existing function delegate to the new query map one to minimize the duplication, but the old function will be short lived. Once the endpoints have been flipped over the prime functions will replace the original